### PR TITLE
virtual_tables: add link_speed column to interface_details for linux

### PR DESCRIPTION
**Pull request by allanliu**
- Add column for speed in Megabits per sec for interface details on Linux.

### CODE REVIEW

**Comment by muffins** [comment]
ok to test

**Comment by allanliu** [comment]
```bash
select * from interface_details where link_speed is not null;
+-------------+-------------------+------+------+--------+-------+----------+----------+------------+-----------+---------+---------+--------+--------+------------+-------------+------------+
| interface   | mac               | type | mtu  | metric | flags | ipackets | opackets | ibytes     | obytes    | ierrors | oerrors | idrops | odrops | collisions | last_change | link_speed |
+-------------+-------------------+------+------+--------+-------+----------+----------+------------+-----------+---------+---------+--------+--------+------------+-------------+------------+
| veth94d3cc1 | a6:ff:98:bd:31:f2 | 1    | 1500 | 0      | 4867  | 92279    | 140164   | 5717392    | 738177862 | 0       | 0       | 0      | 0      | 0          | -1          | 10000      |
| tun0        | 00:00:00:00:00:00 | 254  | 1400 | 0      | 4241  | 1511500  | 996242   | 1735279576 | 325630621 | 0       | 0       | 0      | 3811   | 0          | -1          | 10         |
+-------------+-------------------+------+------+--------+-------+----------+----------+------------+-----------+---------+---------+--------+--------+------------+-------------+------------+
```

**Comment by allanliu** [comment]
Just added a `pci_slot` column to `interface_details` so we can easily make a relationship to the hardware information in `pci_devices`. 
Example:
```
select b.pci_slot, a.vendor, a.model, b.interface from pci_devices a join interface_details b on a.pci_slot = b.pci_slot where a.pci_class = 'Network controller';
+--------------+-------------------+-----------------------------+-----------+
| pci_slot     | vendor            | model                       | interface |
+--------------+-------------------+-----------------------------+-----------+
| 0000:04:00.0 | Intel Corporation | Wireless 8260               | wlp4s0    |
| 0000:00:1f.6 | Intel Corporation | Ethernet Connection I219-LM | enp0s31f6 |
+--------------+-------------------+-----------------------------+-----------+
```

**Comment by obelisk** [comment]
This looks good to merge when 3.2.4 is marked stable.

File: osquery/tables/networking/posix/interfaces.cpp [hunk 1]
```diff
@@ -137,6 +140,30 @@ void genDetailsFromAddr(const struct ifaddrs* addr, QueryData& results) {
         r["type"] = INTEGER_FROM_UCHAR(ifr.ifr_hwaddr.sa_family);
       }
 
+      if (ioctl(fd, SIOCGIFFLAGS, &ifr) >= 0) {
```
Comment (fmanco) [inline osquery/tables/networking/posix/interfaces.cpp]: This block needs to be removed! I guess this came from the merge.

**Comment by allanliu** [comment]
Rebased the branch with head of master.

**Comment by fmanco** [comment]
LGTM thanks for the quick fix.

**Comment by osqueryer** [comment]
:-1: The commit d544eefe23daada307817bce285765774b538870 (Job results: [7123](https://jenkins.osquery.io/job/osqueryPullRequestLinux/7123/)) failed one or more tests (**Linux**).

**Pull request merged**